rfcs/20210415-gpu-only/README.md

```diff
@@ -0,0 +1,185 @@
+# Proposal on introducing GPU only configurations
+
+
+## Motivation
+oneDNN supports multiple CPU and GPU runtimes. While it's possible to build
+oneDNN only for a CPU runtime the GPU runtime always requires a CPU runtime to be
+specified. For some customers the size of the oneDNN library matters a lot therefore
+if the customers only need a GPU runtime then having a redundant CPU runtime
+becomes a problem. One of the customers is OpenVINO whose request is the main motivation
+for introducing an ability to build oneDNN only for GPU runtime.
+
+## Proposal
+The proposal is to introduce the following GPU only configurations:
+* gpu_dpcpp (`DNNL_CPU_RUNTIME=NONE DNNL_GPU_RUNTIME=DPCPP`)
+* gpu_ocl (`DNNL_CPU_RUNTIME=NONE DNNL_GPU_RUNTIME=OCL`)
```

Comment:
At this point we're going to use only 5-6 core primitives from oneDNN, so what do you think about additional parameterization of the build with a list of required primitives?

Reply:
I made some experiments with configuring list of supported primitives at build time. For the experiments I introduced build time options `DNNL_PRIM_LIST` and `DNNL_PROPS_LIST` which could be used in the following way:
`cmake -DDNNL_PRIM_LIST="convolution,matmul,reorder" -DDNNL_PROPS_LIST="fwd" ...`. This will build a configuration that supports the specified primitives for the specified propagation kind.
So the idea is to provide a mechanism that could help you to configure the build from supported functionality perspective.
Would that work you?

+@dzarukin who is working on reducing binary size

Reply:
But in any case, this will be a separate task and likely separate RFC.

Reply:
@densamoilov I think that prim list should contain prop direction, as we'll need convolution backward primitive (which is used for inference), but won't need any other backward primitive.
So something like -DDNNL_PRIM_LIST="convolution_fwd,matmul,reorder" would work perfectly for us.

Reply:
That makes sense, the format of specifying needed functionality can be adjusted to fulfil the needs. I just wanted to share the idea with you.